.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,43 @@
+name: nightly
+on:
+  # This can be used to automatically publish nightlies at UTC nighttime
+  schedule:
+    - cron: '0 7 * * *' # run at 7 AM UTC
+  # This can be used to allow manually triggering nightlies from the web interface
+  workflow_dispatch:
+
+jobs:
+  nightly:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          token: ${{ secrets.GH_NIGHTLY_TOKEN }}
+          fetch-depth: 0
+
+      - run: git fetch --force --tags
+      -
+        name: 'Push new tag'
+        run: |
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+          # A previous release was created using a lightweight tag
+          # git describe by default includes only annotated tags
+          # git describe --tags includes lightweight tags as well
+          DESCRIBE=`git tag -l --sort=-v:refname | grep -v nightly | head -n 1`
+          MAJOR_VERSION=`echo $DESCRIBE | awk '{split($0,a,"."); print a[1]}'`
+          MINOR_VERSION=`echo $DESCRIBE | awk '{split($0,a,"."); print a[2]}'`
+          MINOR_VERSION="$((${MINOR_VERSION} + 1))"
+          TAG="${MAJOR_VERSION}.${MINOR_VERSION}.0-nightly.$(date +'%Y%m%d')"
+          git tag -a $TAG -m "$TAG: nightly build"
+          git push origin $TAG
+      - name: 'Clean up nightly releases'
+        uses: dev-drprasad/delete-older-releases@v0.3.4
```

Comment:
This one is currently the one with the most usage it seems like and it has been abandoned fairly recently. Other alternatives have (seemingly) low usage and are very new:

https://github.com/ShivamHS/delete-tag
https://github.com/marketplace/actions/delete-a-release-by-tag-name
https://github.com/marketplace/actions/delete-stale-releases
https://github.com/marketplace/actions/delete-a-release
https://github.com/larryjoelane/delete-release-action

This is a little more hacky: 
https://github.com/actions/github-script

Reply:
@nimakaviani @cmoulliard Any thoughts here? The GH action maintainers straight up says not to use it for new project so I would rather not use it.

Reply:
I mean, not great that it is not maintained, but other versions are either as old or with very little traction. if it works and since the work is already done, lets use it for now. We switch if things break. Also, it is not on the critical path for the project, so if fixing it is at the cost of losing one or two nightlies, that's fine.

Reply:
I did a quickly review and this one seems good: https://github.com/wow-actions/delete-stale-releases
Why: Their [code](https://github.com/wow-actions/delete-stale-releases/blob/master/src/util.ts) which uses the github action toolkit - https://www.npmjs.com/package/@actions/github and Github Octokit lib: https://octokit.github.io/rest.js/v20#repos-delete-release

Reply:
yes I saw this too. latest commit is from 2 years ago. so is this any better maintained? not sure!

Reply:
https://github.com/actions/github-script doesn't look bad at all. All we need is get releases, parse, put new one in, delete old ones. They even provide the rest client for GH specifically. Published by GH too.

Reply:
We don't even need to put the new one; that's done by goreleaser. We just need to delete the last nightly.

Reply:
alright lets do github-script then.